File: src/main/java/com/example/cqrs/query/dto/AccountDetailResponse.java
package com.example.cqrs.query.dto;

import com.example.cqrs.query.entity.AccountView;
import lombok.*;

/**
 * 계좌 상세 정보에 대한 Response DTO입니다.
 */
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
@Getter
public class AccountDetailResponse {

    private String accountId;
    private double balance;

    // factory method
    public static AccountDetailResponse from(AccountView accountView) {
        return AccountDetailResponse.builder()
                .accountId(accountView.getAccountId())
                .balance(accountView.getBalance())
                .build();
    }

}

File: src/main/java/com/example/cqrs/application/command/account/dto/response/AccountDetailResponse.kt
package com.example.cqrs.application.command.account.dto.response

data class AccountDetailResponse(
    val accountId: String,
    val balance: Double
) {
    companion object {
        fun from(
            accountId: String,
            balance: Double
        ): AccountDetailResponse {
            return AccountDetailResponse(accountId, balance)
        }
    }
}
